src/routes/users.js

```diff
@@ -19,6 +19,24 @@ router.route('/:id').get((req, res) => {
         .catch(err => res.status(400).json('Error: ' + err));
 });
 
+/**
+ * Get user by username
+ */
+ router.route('/:username').get((req, res) => {
+    User.findById(req.params.username)
```

Comment:
Have you tested this? Username is the id in our ER diagram, but I don't see anywhere in mongoose where you linked username to _id of the model. Please refer to login route to search a user by attribute or specify in the model that username is the id.